nbcelltests/__main__.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import argparse
+from .lint import run as runLint
+from .test import run as runTest
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'option',
+        help='Which option to run',
+        default='lint',
+        choices=('lint', 'test'))
+
+    parser.add_argument(
+        'notebook',
+        help='On which notebook to run')
+
+    parser.add_argument(
+        '--lines_per_cell',
+        help='How many lines to allowed per cell',
+        type=int,
+    )
+
+    parser.add_argument(
+        '--cells_per_notebook',
+        help='How many cells are allowed per notebook',
+        type=int,
+    )
+
+    parser.add_argument(
+        '--function_definitions',
+        help='How many function definitions are allowed per notebook',
+        type=int,
+    )
+
+    parser.add_argument(
+        '--class_definitions',
+        help='How many class definitions are allowed per notebook',
+        type=int,
+    )
+
+    parser.add_argument(
+        '--kernelspec',
+        help='Requirements of the kernelspec used in the notebook',
+    )
```

Comment:
@timkpaine This doesn't look right (two things referring to kernelspec here). 

I think kernelspec in the code refers to actual (what's in notebook), while kernelspec_requirements refers to desired (i.e. the rule). In same way that there's cell_count and cells_per_notebook.

No doubt these terms are all a bit confusing/not well named.